src/modules/fancyzones/editor/FancyZonesEditor/Models/CanvasLayoutModel.cs

```diff
@@ -117,46 +117,56 @@ public override LayoutModel Clone()
-                writer.WriteStartObject();
-                writer.WriteString("uuid", "{" + Guid.ToString().ToUpper() + "}");
-                writer.WriteString("name", Name);
+                FileStream outputStream = File.Open(Settings.AppliedZoneSetTmpFile, FileMode.Create);
+                JsonWriterOptions writerOptions = new JsonWriterOptions
+                {
+                    SkipValidation = true,
+                };
+                using (var writer = new Utf8JsonWriter(outputStream, writerOptions))
+                {
+                    writer.WriteStartObject();
+                    writer.WriteString("uuid", "{" + Guid.ToString().ToUpper() + "}");
+                    writer.WriteString("name", Name);
 
-                writer.WriteString("type", "canvas");
+                    writer.WriteString("type", "canvas");
 
-                writer.WriteStartObject("info");
+                    writer.WriteStartObject("info");
 
-                writer.WriteNumber("ref-width", _referenceWidth);
-                writer.WriteNumber("ref-height", _referenceHeight);
+                    writer.WriteNumber("ref-width", _referenceWidth);
+                    writer.WriteNumber("ref-height", _referenceHeight);
 
-                writer.WriteStartArray("zones");
-                foreach (Int32Rect rect in Zones)
-                {
-                    writer.WriteStartObject();
-                    writer.WriteNumber("X", rect.X);
-                    writer.WriteNumber("Y", rect.Y);
-                    writer.WriteNumber("width", rect.Width);
-                    writer.WriteNumber("height", rect.Height);
-                    writer.WriteEndObject();
-                }
+                    writer.WriteStartArray("zones");
+                    foreach (Int32Rect rect in Zones)
+                    {
+                        writer.WriteStartObject();
+                        writer.WriteNumber("X", rect.X);
+                        writer.WriteNumber("Y", rect.Y);
+                        writer.WriteNumber("width", rect.Width);
+                        writer.WriteNumber("height", rect.Height);
+                        writer.WriteEndObject();
+                    }
 
-                writer.WriteEndArray();
+                    writer.WriteEndArray();
 
-                // end info object
-                writer.WriteEndObject();
+                    // end info object
+                    writer.WriteEndObject();
 
-                // end root object
-                writer.WriteEndObject();
-                writer.Flush();
+                    // end root object
+                    writer.WriteEndObject();
+                    writer.Flush();
+                }
+
+                outputStream.Close();
             }
+            catch (Exception ex)
+            {
+                string message = "Please report the bug to https://github.com/microsoft/PowerToys/issues" + "\nError persisting canvas layout: " + ex.Message;

```

Comment:
To avoid code duplication, can we extract these lines outside to a separate function, since we are calling the same part of the code in several places ?

Reply:
Why aren’t we just using an object that is serialized

Reply:
@crutkas 
FZ and FZ editor don't use this json internally, this is only limited to the read/write functions to exchange data between the FZ editor (C#) and the FZ engine (C++).
The data written to the temp file is pretty straightforward and not using an object serialization allows for more flexibility.
Creating a json object here isn't really helping with enforcing a specific format.
We can reconsider it and create a COM object used by both C++ and C# code to read/write these temp files, so any future change to the temp files format will be done in one place (for the read/write) operation (it will still require to update both C++ and C# to support those changes).
In that case we will have to do an estimate of the cost, because it may requires more changes than just move these methods to a COM object.

Reply:
The old code went around this, doing a remote COM server call, that is not possible on ARM.
That was the main reason to switch to the temp files.

Reply:
> To avoid code duplication, can we extract these lines outside to a separate function, since we are calling the same part of the code in several places ?

Done